analyzer/consensus/consensus.go

```diff
@@ -376,6 +433,31 @@ func (m *processor) queueEpochInserts(batch *storage.QueryBatch, data *consensus
 	return nil
 }
 
+// Adapted from https://github.com/oasisprotocol/oasis-core/blob/master/go/consensus/api/transaction/transaction.go#L58
+func unpackTxBody(t *transaction.Transaction) (interface{}, error) {
+	bodyType, ok := bodyTypeForTxMethod[string(t.Method)]
+	if !ok {
+		return nil, fmt.Errorf("unknown method body: %s", t.Method)
+	}
+
+	// Deserialize into correct type.
+	v := reflect.New(reflect.TypeOf(bodyType)).Interface()
+	if err := cbor.Unmarshal(t.Body, v); err != nil {
+		// This may be an older tx from Cobalt; attempt to unmarshal into
+		// the appropriate Cobalt tx type.
+		bodyType, ok = bodyTypeForTxMethodCobalt[string(t.Method)]
+		if !ok {
+			return nil, fmt.Errorf("unable to cbor-decode consensus tx body: %w, body: %s", err, base64.StdEncoding.EncodeToString(t.Body))
+		}
+		v = reflect.New(reflect.TypeOf(bodyType)).Interface()
+		if err = cbor.Unmarshal(t.Body, v); err != nil {
+			return nil, fmt.Errorf("unable to cbor-decode consensus tx body into Cobalt type: %w, body: %s", err, base64.StdEncoding.EncodeToString(t.Body))
```

Comment:
```suggestion
		if err2 = cbor.Unmarshal(t.Body, v); err != nil {
			return nil, fmt.Errorf("unable to cbor-decode consensus tx body into Cobalt type (%w) or newer type (%w), body: %x", err2, err, t.Body)
```

(if we hit an issue, it's more likely to be with post-cobalt decoding as cobalt is set in stone whereas the current version will keep evolving)

Reply:
Updated!